src/hocon_schema.erl

```diff
@@ -324,9 +325,9 @@ field_schema(?UNION(_) = Union, SchemaKey) ->
 field_schema(?ENUM(_) = Enum, SchemaKey) ->
     field_schema(hoconsc:mk(Enum), SchemaKey);
 field_schema(FieldSchema, SchemaKey) when is_function(FieldSchema, 1) ->
-    FieldSchema(SchemaKey);
+    unbox_desc(FieldSchema(SchemaKey));
 field_schema(FieldSchema, SchemaKey) when is_map(FieldSchema) ->
-    maps:get(SchemaKey, FieldSchema, undefined).
+    unbox_desc(maps:get(SchemaKey, FieldSchema, undefined)).
```

Comment:
putting desc resolution here feels too low level and unnecessary expensive as it has to load hocon for each and every filed.

it should be enough to resolve it in hocon_schema_json and hocon_schema_md.

Reply:
All needs desc should be resolved.
emqx swagger's desc also need.
if we expose the resolve function to the user. user must call:
```erlang
Desc = hocon_schema:field_schema(Hocon,desc),
hocon_schema:resolve_desc(Desc).
```
it only resolves the desc field, it is not expensive for other fields(we only match `?DESC`)

Reply:
but the other fields call `resolve_desc/1` if not necessary.
I want to introduce a new function called hocon_schema:desc/1.
```erlang
hocon_schema:desc(Hocon) ->
Desc = hocon_schema:field_schema(Hocon,desc),
hocon_schema:resolve_desc(Desc).
```
If you want to get the desc, no longer use `field_schema/2` directly.

Reply:
OK.
But maybe `resolve_schema(ResolverFun)`, and `ResolverFun` can have a cache implemented inside.